src/test/java/spoon/test/filters/FilterTest.java

```diff
@@ -1124,4 +1125,17 @@ public void accept(CtType<?> ctType) {
 		// only one subtype remains unvisited
 		assertEquals(1, c2.counter);
 	}
+
+	@Test
+	public void testNameFilterWithGenericType() {
+		// contract: NameFilter of T should only return T elements
+
+		Launcher spoon = new Launcher();
+		spoon.addInputResource("./src/test/java/spoon/test/imports/testclasses/internal4/Constants.java");
+		spoon.buildModel();
+
+		CtType type = spoon.getFactory().Type().get(Constants.class);
+		List<CtMethod> ctMethods = type.getElements(new NameFilter<CtMethod>("CONSTANT"));
```

Comment:
I cannot check it now, but I would say that it newer worked different. The NameFilter checks the name only. If one wants to filter by type s/he has to use TypeFilter. So it is not a bug of query engine, but old behavior of namefilter.
Anaway as client I would like the new constructor, which allows to filter by name and by type too - it is often the case...